api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -539,10 +539,18 @@ public static NestedField required(int id, String name, Type type, String doc) {
       return new NestedField(false, id, name, type, doc, null, null);
     }
 
+    /**
+     * @deprecated will be removed in 2.0.0; use builder() instead.
```

Comment:
nit: we typically try to use javadoc links here when possible
```suggestion
     * @deprecated will be removed in 2.0.0; use {@link #builder()} instead.
```